lib/webauthn/authenticator_attestation_response.rb

```diff
@@ -86,62 +81,15 @@ def type
     end
 
     def valid_attested_credential?
-      authenticator_data.attested_credential_data_included? &&
-        authenticator_data.attested_credential_data.valid?
+      attestation.valid_attested_credential?
     end
 
     def valid_attestation_statement?
-      @attestation_type, @attestation_trust_path = attestation_statement.valid?(authenticator_data, client_data.hash)
+      @attestation_type, @attestation_trust_path = attestation.valid_attestation_statement?(client_data.hash)
     end
 
     def valid_attestation_trustworthiness?
-      case @attestation_type
-      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_NONE
-        WebAuthn.configuration.acceptable_attestation_types.include?('None')
-      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_SELF
-        WebAuthn.configuration.acceptable_attestation_types.include?('Self')
-      else
-        WebAuthn.configuration.acceptable_attestation_types.include?(@attestation_type) &&
-          attestation_root_certificates_store.verify(leaf_certificate, signing_certificates)
-      end
-    end
-
-    def raw_subject_key_identifier(certificate)
-      extension = certificate.extensions.detect { |ext| ext.oid == "subjectKeyIdentifier" }
-      return unless extension
-
-      ext_asn1 = OpenSSL::ASN1.decode(extension.to_der)
-      ext_value = ext_asn1.value.last
-      OpenSSL::ASN1.decode(ext_value.value).value
-    end
-
-    def attestation_root_certificates_store
-      certificates =
-        WebAuthn.configuration.attestation_root_certificates_finders.reduce([]) do |certs, finder|
-          if certs.empty?
-            finder.find(
-              attestation_format: attestation_format,
-              aaguid: aaguid,
-              attestation_certificate_key_id: attestation_certificate_key
-            ) || []
-          else
-            certs
-          end
-        end
-
-      OpenSSL::X509::Store.new.tap do |store|
-        certificates.each do |cert|
-          store.add_cert(cert)
-        end
-      end
-    end
-
-    def signing_certificates
-      @attestation_trust_path[1..-1]
-    end
-
-    def leaf_certificate
-      @attestation_trust_path.first
```

Comment:
Kind of a signal that this was asking for encapsulation was the fact that a lot of the variable names in the above deleted lines were namespaced with "attestation" word over and over.